File: 프로그래머스/2/42626. 더 맵게/더 맵게.java
import java.io.*;
import java.util.*;

class Solution {
    public int solution(int[] scoville, int K) {
        int answer = 0;
        PriorityQueue<Integer> pq = new PriorityQueue<>();
        
        for(int s : scoville) {
            pq.offer(s);
        }
        
        while(pq.peek() < K && pq.size() > 1) {
            int s1 = pq.poll();
            int s2 = pq.poll();    
            pq.offer(s1 + s2 * 2);
            answer++;
        }
        
        return pq.peek() < K ? -1 : answer;
    }
}

File: 프로그래머스/2/42626. 더 맵게/더 맵게.cpp
#include <bits/stdc++.h>
using namespace std;

int solution(vector<int> scoville, int K) {
    int answer = 0;
    priority_queue<int, vector<int>, greater<int>> pq;
    
    for(int s : scoville) {
        pq.push(s);
    }
    
    while(pq.top() < K && pq.size() > 1) {
        int s1 = pq.top();
        pq.pop();
        
        int s2 = pq.top();
        pq.pop();
        
        pq.push(s1 + s2 * 2);
        answer++;
    }
    
    return pq.top() < K ? -1 : answer;
}